__device-tests__/gutenberg-editor-writing-flow.test.js

```diff
@@ -0,0 +1,127 @@
+			const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.paragraph
+			);
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphText[ 0 ]
+			);
+
+			// Toggle various formatting options, then proceed with typing more text
+			await editorPage.toggleFormatting( 'Bold' );
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphText[ 1 ]
+			);
+
+			await editorPage.toggleFormatting( 'Italic' );
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphText[ 2 ]
+			);
+
+			await editorPage.toggleFormatting( 'Italic' );
+			await editorPage.toggleFormatting( 'Bold' );
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphText[ 3 ]
+			);
+
+			await editorPage.toggleFormatting( 'Strikethrough' );
+
+			// TODO: Invoke assertion or snapshot here?
+
+			await editorPage.removeBlock();
+		} );
+	} );
+
+	describe( `TC009 - Test autocorrection doesn't apply formatting to Heading`, () => {
+		it( 'checks that formatting is not applied to autocorrected text', async () => {
+			const headingText = `Mispelled`;
+
+			// Add a Heading block without any formatting applied.
+			await editorPage.addNewBlock( blockNames.heading );
+			const headingBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.heading
+			);
+
+			// Type a sentence with a word misspelled that will be autocorrected by the editor
+			await editorPage.typeTextToTextBlock(
+				headingBlockElement,
+				headingText
+			);
+
+			// Highlight the word that was autocorrected
+			await tapSelectAllAboveElement(
+				editorPage.driver,
+				headingBlockElement
+			);
+
+			// Check that the word has no formatting applied
+			const selectedText = await editorPage.getTextForParagraphBlockAtPosition(
+				1
+			);
+			expect( selectedText ).toMatch( headingText );
+
+			await editorPage.removeBlock();
+		} );
+	} );
+
+	describe( `TC010 - Test autocorrection doesn't remove formatting from Heading`, () => {
+		it( 'checks that formatting is not removed from autocorrected text', async () => {
+			const headingText = 'Mispelled';
+
+			// Add a Heading block
+			await editorPage.addNewBlock( blockNames.heading );
+			const headingBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.heading
+			);
+
+			// Toggle bold formatting
+			// Type a sentence with a word misspelled that will be autocorrected by the editor
+
+			await editorPage.toggleFormatting( 'Bold' );
+			await editorPage.typeTextToTextBlock(
+				headingBlockElement,
+				headingText
+			);
+
+			// Highlight the word that was autocorrected
+			await tapSelectAllAboveElement(
+				editorPage.driver,
+				headingBlockElement
+			);
+
+			// Check that the word has formatting applied
+			const selectedText = await editorPage.getTextForParagraphBlockAtPosition(
+				1
+			);
+			expect( selectedText ).toMatch( headingText );
```

Comment:
Similar to my other comment, it may make most sense to assert the output of [`getHtmlContent`](https://github.com/WordPress/gutenberg/blob/dfdfaa3135b32af9018a41b832c2538c49adf747/packages/react-native-editor/__device-tests__/pages/editor-page.js#L261-L269) instead. 

This currently asserts that `selectedText` matches `headingText` (`'Mispelled'`). The latter is a plain text string, but the test description states we expected formatting applied.

Reply:
Noting that we should apply [the planned use of `getHtmlContent`](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5610#discussion_r1164201444) here as well.